pyiceberg/partitioning.py

```diff
@@ -222,11 +223,14 @@ def partition_type(self, schema: Schema) -> StructType:
         :return: A StructType that represents the PartitionSpec, with a NestedField for each PartitionField.
         """
         nested_fields = []
+        schema_ids = index_by_id(schema)
         for field in self.fields:
-            source_type = schema.find_type(field.source_id)
-            result_type = field.transform.result_type(source_type)
-            required = schema.find_field(field.source_id).required
-            nested_fields.append(NestedField(field.field_id, field.name, result_type, required=required))
+            if field.source_id in schema_ids:
+                source_field = schema.find_field(field.source_id)
```

Comment:
Small suggestion to simplify this:
```suggestion
            if source_field := index_by_id.get(field.source_id):
```

Reply:
nice, great catch thanks